soc/espressif/esp32/esp32-mp.c

```diff
@@ -453,4 +453,10 @@ int esp_appcpu_init(void)
 
 	return 0;
 }
+
+#if !defined(CONFIG_MCUBOOT)
+extern int esp_appcpu_init(void);
+SYS_INIT(esp_appcpu_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
use soc hook

Reply:
I did not get it..  do you mean adding `CONFIG_SOC_ENABLE_APPCPU`?

Reply:
in `include/zephyr/platform/hooks.h`:
```
/**
 * @brief SoC hook executed before the kernel and devices are initialized.
 *
 * This hook is implemented by the SoC and can be used to perform any
 * SoC-specific initialization.
 */
void soc_early_init_hook(void);
```
Enabled with `CONFIG_SOC_EARLY_INIT_HOOK`

Reply:
That `SYS_INIT` call above was just moved from `soc.c` into `esp32-mp.c` where it belongs. Do you mind if I handle this in another PR as you suggested?

Reply:
@marekmatej, would you update https://github.com/zephyrproject-rtos/zephyr/pull/85514 with the recommendation above if needed?

Reply:
Sure, I'll replace the use of `SYS_INIT` in my AMP PRs. Can we leave this as is for a while @nordicjm ?

Reply:
yes